iceoryx_posh/source/runtime/posh_runtime_impl.cpp

```diff
@@ -648,7 +653,7 @@ popo::ConditionVariableData* PoshRuntimeImpl::getMiddlewareConditionVariable() n
 bool PoshRuntimeImpl::sendRequestToRouDi(const IpcMessage& msg, IpcMessage& answer) noexcept
 {
     // runtime must be thread safe
-    std::lock_guard<posix::mutex> g(m_appIpcRequestMutex);
+    std::lock_guard<posix::mutex> g(m_appIpcRequestMutex.value());
```

Comment:
Do we have the guarantee that the `optional` always contains a value?

Reply:
This reminds me of the discussions on the thread implementation...

Reply:
I think I'll create a `Discussion` instead and create the issue only if there is agreement on something like this